samples/kotlin-android-app-destinations/src/test/java/com/segment/analytics/destinations/plugins/MixpanelDestinationTests.kt

```diff
@@ -0,0 +1,902 @@
+            )
+        }
+    }
+
+    @Test
+    fun `group is handled correctly`() {
+        val sampleEvent = GroupEvent(
+            groupId = "grp-123",
+            traits = buildJsonObject { put("company", "abc") }
+        ).apply {
+            messageId = "qwerty-1234"
+            anonymousId = "anonId"
+            integrations = emptyJsonObject
+            context = emptyJsonObject
+            timestamp = "2021-07-13T00:59:09"
+        }
+        mixpanelDestination.setupTest(
+            MixpanelSettings(
+                token = "",
+            )
+        )
+        val groupEvent = mixpanelDestination.group(sampleEvent)
+
+        assertNotNull(groupEvent)
+
+        // check mixpanel getGroup called with groupKey default "[Segment] Group" and groupID "grp-123"
+        verify { mockMixpanel.getGroup("[Segment] Group", "grp-123") }
+
+        // verify to see that the same Traits passed in the integration
+        // transformed to a JsonObject are setOnce on the Group object
+        verify {
+            mockMixpanelGroup.setOnce(
+                matchJsonObject(
+                    JSONObject().put("company", "abc")
+                )
+            )
+        }
+
+        // groupKey defaults to groupId, since no name is set
+        verify { mockMixpanel.setGroup("[Segment] Group", "grp-123") }
+    }
+
+    @Test
+    fun `group with group name`() {
+        val sampleEvent = GroupEvent(
+            groupId = "grp-123",
+            traits = buildJsonObject {
+                put("name", "ABC network")
+                put("company", "abc")
+            }
+        ).apply {
+            messageId = "qwerty-1234"
+            anonymousId = "anonId"
+            integrations = emptyJsonObject
+            context = emptyJsonObject
+            timestamp = "2021-07-13T00:59:09"
+        }
+        mixpanelDestination.setupTest(
+            MixpanelSettings(
+                token = "",
+            )
+        )
+        val groupEvent = mixpanelDestination.group(sampleEvent)
+
+        assertNotNull(groupEvent)
+
+        // check mixpanel getGroup called with groupKey default "ABC network" and groupID "grp-123"
+        verify { mockMixpanel.getGroup("ABC network", "grp-123") }
+        // verify to see that the same Traits passed in the integration
+        // transformed to a JsonObject are setOnce on the Group object
+        verify {
+            mockMixpanelGroup.setOnce(
+                matchJsonObject(
+                    JSONObject()
+                        .put("name", "ABC network")
+                        .put("company", "abc")
+                )
+            )
+        }
+
+        // groupKey defaults to groupId, since no name is set
+        verify { mockMixpanel.setGroup("ABC network", "grp-123") }
+    }
+
+    /* PRIVATE TEST FUNCTIONS */
+
+    private fun MixpanelDestination.setupTest(mixpanelSettings: MixpanelSettings) {
+        this.mixpanel = mockMixpanel
```

Comment:
same as this one. probably can use the way of mock private properties for setup, and keep these properties as private